src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/task.py

```diff
@@ -0,0 +1,469 @@
+        cmd.cli_ctx, registry_name, resource_group_name, TASK_NOT_SUPPORTED)
+
+    from ._client_factory import cf_acr_registries
+    client_registries = cf_acr_registries(cmd.cli_ctx)
+
+    queued_run = LongRunningOperation(cmd.cli_ctx)(
+        client_registries.schedule_run(
+            resource_group_name,
+            registry_name,
+            TaskRunRequest(
+                task_name=task_name,
+                values=(set_value if set_value else []) + (set_secret if set_secret else [])
+            )
+        )
+    )
+
+    run_id = queued_run.run_id
+    logger.warning("Queued a run with ID: %s", run_id)
+
+    if no_wait:
+        return queued_run
+
+    logger.warning("Waiting for an agent...")
+
+    if no_logs:
+        return get_run_with_polling(client, run_id, registry_name, resource_group_name)
+
+    return stream_logs(client, run_id, registry_name, resource_group_name, True)
+
+
+def acr_task_show_run(cmd,
+                      client,  # cf_acr_runs
+                      run_id,
+                      registry_name,
+                      resource_group_name=None):
+    _, resource_group_name = validate_managed_registry(
+        cmd.cli_ctx, registry_name, resource_group_name, TASK_NOT_SUPPORTED)
+    return client.get(resource_group_name, registry_name, run_id)
+
+
+def acr_task_cancel_run(cmd,
+                        client,  # cf_acr_runs
+                        run_id,
+                        registry_name,
+                        resource_group_name=None):
+    _, resource_group_name = validate_managed_registry(
+        cmd.cli_ctx, registry_name, resource_group_name, TASK_NOT_SUPPORTED)
+    return client.cancel(resource_group_name, registry_name, run_id)
+
+
+def acr_task_list_runs(cmd,
+                       client,  # cf_acr_runs
+                       registry_name,
+                       top=15,
+                       task_name=None,
+                       run_status=None,
+                       image=None,
+                       resource_group_name=None):
+    _, resource_group_name = validate_managed_registry(
+        cmd.cli_ctx, registry_name, resource_group_name, TASK_NOT_SUPPORTED)
+
+    filter_str = None
+    filter_str = _add_run_filter(filter_str, 'TaskName', task_name, 'eq')
+    filter_str = _add_run_filter(filter_str, 'Status', run_status, 'eq')
+
+    if image:
+        from .repository import get_image_digest
+        try:
+            repository, _, manifest = get_image_digest(cmd.cli_ctx, registry_name, resource_group_name, image)
+            filter_str = _add_run_filter(
+                filter_str, 'OutputImageManifests', '{}@{}'.format(repository, manifest), 'contains')
+        except CLIError as e:
+            raise CLIError("Could not find image '{}'. {}".format(image, e))
+
+    return client.list(resource_group_name, registry_name, filter=filter_str, top=top)
+
+
+def _add_run_filter(orig_filter, name, value, operator):
```

Comment:
NIT: quite some duplicated code from build_task.py

Reply:
I would keep this, and nuke the `build_task.py` file in the next iteration when we deprecate `build-task`. Don't want to create a `_build_task_utils.py`